dynamic-data-masking/macros/apply_data_masking.sql

```diff
@@ -0,0 +1,12 @@
+{% macro apply_data_masking(columns) %}
+{% set relation_type='view' if model.config.materialized == 'view' else 'table' %}
```

Comment:
I'd think about generalizing this macro to take two arguments, `relation` + `columns`. Then:
```
{% set rel = adapter.get_relation(relation.database, relation.schema, relation.identifier) %}
```
which enables you to use `rel.type` instead of the `config.materialized` conditional.

Then you could have `this` be the default arg to relation, or call it below as
```sql
post_hook=apply_data_masking(
            relation=this,
            columns=['first_name', 'last_name']
        )
```

just a thought!

Reply:
I was thinking along the same lines, however when I change this like:
```
alter {{ relation.type }} ...
```
it compiles to `None`

Transaction behavior? Maybe because we [created the ref from a node](https://github.com/fishtown-analytics/dbt/blob/dev/kiyoshi-kuromiya/core/dbt/adapters/base/relation.py#L222) and there doesn't seem to be a `type` param?

Reply:
Hmm looks like `adapter.get_relation` isn't working in a post-hook the way I'd hoped. Thanks for humoring me; your original code works well!